### PR TITLE
Adding ranges information on the file browser

### PR DESCRIPTION
**Pull request by markccchiang**
#845, adding ranges information in the file preview. Including RA, Dec, Frequency, Velocity, and Stokes coverages. As shown below:
截圖 2022-02-22 上午10 37 32
 
@kswang1029 do we also need ranges information for compressed FITS files in the file preview? If so, I will do it in another separate PR.

### CODE REVIEW

**Comment by kswang1029** [comment]
> #845, adding ranges information in the file preview. Including RA, Dec, Frequency, Velocity, and Stokes coverages. As shown below: 截圖 2022-02-22 上午10 37 32
> 
> @kswang1029 do we also need ranges information for compressed FITS files in the file preview? If so, I will do it in another separate PR.

Yes, for compressed fits we need this set of info too. Good idea to support it in a separate PR.

**Comment by veggiesaurus** [comment]
This is looking good. Do you think we can just use RA and DEC shorthands? I also wonder what we should do for situations where the image is in galactic coordinates. Should we not report the spatial range in GLON/GLAT?

**Comment by kswang1029** [comment]
> This is looking good. Do you think we can just use RA and DEC shorthands? I also wonder what we should do for situations where the image is in galactic coordinates. Should we not report the spatial range in GLON/GLAT?

Depending on what CTYPE1/2/3 says we should have dynamic naming conventions.

File: src/FileList/FileExtInfoLoader.cc [hunk 1]
```diff
@@ -1175,56 +1176,65 @@ void FileExtInfoLoader::GetCoordNames(std::string& ctype1, std::string& ctype2,
...
+            x_range_vec[1] = world[0];
+            std::string x_end = direction_coord.format(units, casacore::Coordinate::DEFAULT, world[0], 0, true, true);
+            y_range_vec[1] = world[1];
+            std::string y_end = direction_coord.format(units, casacore::Coordinate::DEFAULT, world[1], 1, true, true);
+
+            // Set x and y coordinate names
+            if (axis_names(0) == "Right Ascension") {
+                axis_names(0) = "RA";
+            } else if (axis_names(0) == "Longitude") {
+                axis_names(0) = "GLON";
```
Comment (kswang1029) [inline src/FileList/FileExtInfoLoader.cc]: just LON and LAT. 

G means galactic. There is also ecliptic coordinate as ELON and ELAT. So let just try to display them as LON and LAT for simplicity.

**Review by kswang1029 — APPROVED** [review]
Tested and it looks good. The current implementation also fixes a bug in the casaviewer where the reported spatial coverage has one extra pixel offset. 👍

**Comment by kswang1029** [comment]
@confluence For the support of compressed FITS, it could be in another PR if you prefer.

**Comment by markccchiang** [comment]
> @confluence For the support of compressed FITS, it could be in another PR if you prefer.

For compressed FITS, the range information can be seen after opening the file. But it can not show on the preview stage. In order to keep the performance, I guess it is better not to open the file on the preview stage. We have to use another way to calculate the ranges from the header information.

**Review by pford — APPROVED** [review]

**Review by confluence — APPROVED** [review]

**Pull request merged**